### PR TITLE
Update status for {DateTime,Number}Format.formatToParts

### PR DESCRIPTION
**Pull request by jungshik**
Please, take a look. This is to update the support status of {DateTime,Number}Format.formatToParts in Chrome.

### CODE REVIEW

File: javascript/builtins/Intl.json [hunk 1]
```diff
@@ -708,14 +708,21 @@
                   "version_added": false
                 },
                 "chrome": {
```
Comment (Elchi3) [inline javascript/builtins/Intl.json]: If there are multiple version ranges here, this needs to be an array instead of an object. Travis is failing because the JSON is invalid here.

File: javascript/builtins/Intl.json [hunk 2]
```diff
@@ -708,14 +708,21 @@
...
-                    "name": "--datetime-format-to-parts"
-                  }
+                  {
+                    "version_added": "57"
+                  },
+                  {
+                    "version_added": "55",
+                    "version_removed": "60",
+                    "flag": {
+                      "type": "preference",
```
Comment (Elchi3) [inline javascript/builtins/Intl.json]: Why did you change this from compile_flag to preference? I think it was a compile_flag.

File: javascript/builtins/Intl.json [hunk 3]
```diff
@@ -708,14 +708,21 @@
...
+                      "name": "--datetime-format-to-parts"
+                    }
+                  },
+                  "notes": "Enabled by default in 57. The flag was removed in 60"
```
Comment (Elchi3) [inline javascript/builtins/Intl.json]: This note is redundant, please remove it.

File: javascript/builtins/Intl.json [hunk 4]
```diff
@@ -1082,6 +1089,70 @@
...
+            "__compat": {
+              "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/formatToParts",
+              "support": {
+                "webview_android": {
+                  "version_added": false
+                },
+                "chrome": {
+                  "version_added": "63",
+                  "flag": {
+                    "type": "preference",
```
Comment (Elchi3) [inline javascript/builtins/Intl.json]: Should be `compile_flag`, I guess, if it is not a real preference but controllable using the commandline.

File: javascript/builtins/Intl.json [hunk 5]
```diff
@@ -1082,6 +1089,70 @@
...
+                  "notes": "Need to set the flag on the commandline via --js-flag"
+                },
+                "edge": {
+                  "version_added": null
+                },
+                "edge_mobile": {
+                  "version_added": null
+                },
+                "firefox": {
+                  "version_added": null
```
Comment (Elchi3) [inline javascript/builtins/Intl.json]: It's in Firefox 58, see https://bugzilla.mozilla.org/show_bug.cgi?id=1403319

File: javascript/builtins/Intl.json [hunk 6]
```diff
@@ -1082,6 +1089,70 @@
...
+                },
+                "safari": {
+                  "version_added": null
+                },
+                "safari_ios": {
+                  "version_added": null
+                }
+              },
+              "status": {
+                "experimental": true,
```
Comment (Elchi3) [inline javascript/builtins/Intl.json]: You changed Intl.DateTimeFormat.prototype.formatToParts to be longer experimental, I guess Intl.NumberFormat.prototype.formatToParts should then also not be experimental, or am I wrong?

File: javascript/builtins/Intl.json (hunk 2 above)
Comment (jungshik) [reply javascript/builtins/Intl.json]: No, it's not a compile flag but a runtime flag that can be specified like this: 

chrome --js-flags="--datetime-format-to-parts"  

In v8 standalone, 

d8 --datetime-format-to-parts 

If there were a separate enum for 'runtime-flag', I'd use it, but there are only two, 'preference' and 'compile-time flag'. So, I have to choose the former.

File: javascript/builtins/Intl.json (hunk 4 above)
Comment (jungshik) [reply javascript/builtins/Intl.json]: Yes, it's controlled by a command line flag, but it's not a compile time flag, either. If I have to choose between the two, it's closer to 'preferecne' because it can be controlled at run-time. 

The best solution is to add a 3rd enum for flag type: 'command line flag'.

File: javascript/builtins/Intl.json (hunk 6 above)
Comment (jungshik) [reply javascript/builtins/Intl.json]: The former is a part of Ecma 402 2017, but the latter is not yet as far as I can tell.

File: javascript/builtins/Intl.json (hunk 5 above)
Comment (jungshik) [reply javascript/builtins/Intl.json]: Good to know. I'll change that.

File: javascript/builtins/Intl.json (hunk 3 above)
Comment (jungshik) [reply javascript/builtins/Intl.json]: I can, but without notes, I'm afraid it's not clear in what version it's controlled by a flag and in what version it's enabled unconditionally.

File: javascript/builtins/Intl.json (hunk 1 above)
Comment (jungshik) [reply javascript/builtins/Intl.json]: Ok. thank you. I'll fix that.

**Comment by jungshik** [comment]
Thank you for comments. I'll fix failures. For other issues, please see my replies inline.

File: javascript/builtins/Intl.json (hunk 2 above)
Comment (Elchi3) [reply javascript/builtins/Intl.json]: Thanks for clarifying, I agree we need runtime flags as an option in the schema. I've opened https://github.com/mdn/browser-compat-data/pull/615 for this.

File: javascript/builtins/Intl.json (hunk 3 above)
Comment (Elchi3) [reply javascript/builtins/Intl.json]: We're generating some notes from the data and we're experimenting with a new compat table layout to address this. Did you see our beta tables? https://discourse.mozilla.org/t/beta-testing-new-compatability-tables/21269

File: javascript/builtins/Intl.json (hunk 4 above)
Comment (Elchi3) [reply javascript/builtins/Intl.json]: I agree to add a 3rd enum and did so in PR #615 .

File: javascript/builtins/Intl.json (hunk 6 above)
Comment (Elchi3) [reply javascript/builtins/Intl.json]: Okay, I didn't dig too deep in the specs, thanks!

**Comment by Elchi3** [comment]
Thanks! Almost good to go.

The following

```json
"chrome": {
   "version_added": "55",
   "flag": {
     "type": "runtime_flag",
     "name": "--datetime-format-to-parts"
   },
   "notes": "Enabled by default in 57. The flag was removed in 60"
 },
```

can be expressed as two support ranges:

```json
"chrome": [
  {
    "version_added": "57"
  },
  {
    "version_added": "55",
    "version_removed": "60",
    "flag": {
      "type": "runtime_flag",
      "name": "--datetime-format-to-parts"
    }
  }
],
```

**Comment by Elchi3** [comment]
I've created a page for NumberFormat.formatToParts in the meantime: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/formatToParts

**Comment by Elchi3** [comment]
I fixed the remaining issues in a new PR on top of this one https://github.com/mdn/browser-compat-data/pull/690
